File: fixtures/scale/ruby/large/lib/check_nibblexor.rb
require_relative 'nibblexor_encrypt'
require_relative 'nibblexor_decrypt'
require_relative 'corpus_mask'

def check_nibblexor
  corpus_mask.each do |text|
    sealed = nibblexor_encrypt(text, 6)
    opened = nibblexor_decrypt(sealed, 6)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_nibblexor.py
from cipherlab.ciphers.nibblexor import nibblexor_encrypt, nibblexor_decrypt
from cipherlab.specs.nibblexor_spec import nibblexor_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_nibblexor():
    spec = nibblexor_spec()
    for text in corpus_mask():
        sealed = nibblexor_encrypt(text, spec["key"])
        opened = nibblexor_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"
